tests/test_metric_reffree.py

```diff
@@ -0,0 +1,77 @@
+import biotite.interface.rdkit as rdkit_interface
+import pytest
+from rdkit import Chem
+from peppr.metric_reffree import (
+    LigandValenceViolations,
+    _count_valence_violations,
+    _count_valence_violations_atomarray,
+)
+
+
+def _test_mols() -> dict[Chem.Mol, int]:
+    good_smiles = [
+        "C1=CC=CC=C1",
+        "C1=CC=CC=C1[H]",
+        "N(=O)=O",
+        "C=P(=O)O",
+        "O=Cl(=O)O",
+        "OC1C(COP(=O)(OP(=O)(OP(=O)(O)O)O)O)OC(C1O)n1cnc2c1ncnc2N",
+    ]
+
+    one_viol = [
+        "C[C@H](FC)C",
+        "CO(C)C",  # This probably shouldn't read as 0 violations, but it does currently.
+        # TODO: examine `sanitize` to determine whether needs a fix.
+        "CF(C)C",
+    ]
+
+    two_viols = ["NC(O)NCCC1:C:C:C:C([C@H](C=F)/C=O/C(=O)(O)O):C:1"]
+
+    params = Chem.SmilesParserParams()
+    params.sanitize = False
+    good_mols = {Chem.MolFromSmiles(smile, params=params): 0 for smile in good_smiles}
+    one_viol = {Chem.MolFromSmiles(smile, params=params): 1 for smile in one_viol}
+    two_viols = {Chem.MolFromSmiles(smile, params=params): 2 for smile in two_viols}
+    return good_mols | one_viol | two_viols
+
+
+@pytest.mark.parametrize(
+    ["mol", "expected_viols"],
+    [(mol, num_exp_viols) for mol, num_exp_viols in _test_mols().items()],
+)
+def test_count_valence_violations(mol, expected_viols):
```

Comment:
Could you add docstrings to these test functions, that briefly explain what should be tested and how it is done?
```suggestion
def test_count_valence_violations(mol, expected_viols):
"""
Check if the correct number of valence violations are found by
:class:`LigandValenceViolations` by checking the metric against molecules
with a known number of violations.
"""
```